Deprecate google_firebase_project_location (#8301)
GoogleCloudPlatform/magic-modules: mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown

### `deletion_policy` now defaults to `DELETE`

Previously, `google_firebase_web_app` deletions default to `ABANDON`, which means to only stop tracking the WebApp in Terraform. The actual app is not deleted from the Firebase project. If you are relying on this behavior, set `deletion_policy` to `ABANDON` explicitly in the new version.

> ## Resource: `google_firebase_project_location`

Reviewer: This should come before the previous resource to maintain alphabetical order

Author: Done

> ### Deprecated

Reviewer: ```suggestion
### `google_firebase_project_location` is now removed
```

> The concept of a default location is no longer applicable. Consult the supported locations

Reviewer: After looking over the docs for a while, I think this might be a better way to introduce the difference (similar to [this note from the 4.0 release](https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/version_3_upgrade#ip_allocation_policy-will-catch-out-of-band-changes-use_ip_aliases-removed).)

```suggestion
In `4.X`, `google_firebase_project_location` would implicitly trigger creation of an App Engine application with a default Cloud Storage bucket, located in the specified `location_id`. With `5.0.0`, all resources should instead be set up explicitly.
```

Does that sound accurate? If so, I'd recommend also incorporating language from [this other note from 4.0 related to halting implicit resource creation](https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/version_3_upgrade#automatic-subnetwork-creation-for-vpc-native-clusters-removed). In particular, the guide should mention that the implicitly-created resources need to be imported.

Author: IC, let me specify the migration path.

### `liveness_probe.tcp_socket` is now removed

This unsupported field was introduced incorrectly. It is now removed.

## Resource: `google_firebase_project_location`

### `google_firebase_project_location` is now removed

In `4.X`, `google_firebase_project_location` would implicitly trigger creation of an App Engine application with a default Cloud Storage bucket and Firestore database, located in the specified `location_id`. With `5.0.0`, these resources should instead be set up explicitly.

> - If you only used `google_firebase_project_location` to create a default Storage bucket, 

Reviewer: I think that it would probably be fine to remove the content from here to the `Upgrade instructions` header - this was previously the "Detailed changes section. There aren't really any changes to the resources, just removing the implicit creation, and the information covered here is better covered in the Upgrade Instructions section.

If there's anything important that you think would be lost, could it be integrated into the Upgrade Instructions section instead?

Author: My understanding of the `Upgrade instructions` section is how to edit an existing terraform config to be usable in 5.x.x. However, the list here is more geared towards how to create new resources going forward without an existing/old resource.

Reviewer: Hm, I see. I'd prefer to keep the upgrade guide focused on the upgrade steps, but giving guidance on how to create new resources going forward makes sense. What would you think about moving the "For more information on configuring Firebase resources with Terraform" line up to this point to cover the "new resources going forward" information instead of writing out this list?

Author: Good idea. How about now?

resource "google_firebase_storage_bucket" "default-bucket" {
  provider  = google-beta
  project   = google_firebase_project.default.project
  bucket_id = google_app_engine_application.default.default_bucket
}
```
- If you only used `google_firebase_project_location` to create a Firestore instance, use
  `google_firestore_database` instead and specify the location in the resource block.
- If you use `google_firebase_project_location` for both a default Storage bucket and Firestore:
  - Set `database_type = "CLOUD_FIRESTORE"` in your `google_app_engine_application`
  - Add the `google_firestore_database` resource to the `depends_on` block in `google_app_engine_application`
- If you use App Engine intentinally, make sure `google_app_engine_application` and `google_firestore_database`
  are in compatible locations. A quick way to test is to create `google_firestore_database` first, and
  `google_app_engine_application` will fail if it's in an incompatible location.

#### Upgrade instructions

If you have existing resources created using `google_firebase_project_location`:
1. Remove the `google_firebase_project_location` block

> 1. Add `google_app_engine_application` and `google_firebase_storage_bucket` if applicable

Reviewer: Since these two resources and google_firestore_database are all "if applicable", would it make sense to include all the resources here and all the imports in the next list item?

```suggestion
1. Add blocks according to "New config" in this section for any of the following that you need: `google_app_engine_application`, `google_firebase_storage_bucket`, and/or `google_firestore_database`.
```

> 1. Import the existing App Engine application and Storage bucket

Reviewer: ```suggestion
1. Import the existing resources corresponding to the blocks added in the previous step:
```

1. Import the existing App Engine application and Storage bucket
   `terraform import google_app_engine_application.default <project-id>`

>    `terraform import google_firebase_storage_bucket.default-bucket <project-id>/<project.id>.appspot.com`

Author: Oops. Wrong punctuation used.

> 1. Add `google_firestore_database` if applicable

Reviewer: ```suggestion
```

> 1. Import the existing Firestore database

Reviewer: ```suggestion
```
